pkg/k8s/watchers/node.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package watchers
+
+import (
+	"github.com/cilium/cilium/pkg/comparator"
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/informer"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/core/v1"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (k *K8sWatcher) nodesInit(k8sClient kubernetes.Interface) {
+	_, nodeController := informer.NewInformer(
+		cache.NewListWatchFromClient(k8sClient.CoreV1().RESTClient(),
+			"nodes", v1.NamespaceAll, fields.ParseSelectorOrDie("metadata.name="+nodeTypes.GetName())),
+		&v1.Node{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				var valid, equal bool
+				if oldNode := k8s.ObjToV1Node(oldObj); oldNode != nil {
+					valid = true
+					if newNode := k8s.ObjToV1Node(newObj); newNode != nil {
+						oldNodeLabels := oldNode.GetLabels()
+						newNodeLabels := newNode.GetLabels()
+						if comparator.MapStringEquals(oldNodeLabels, newNodeLabels) {
+							equal = true
+						} else {
+							err := k.updateK8sNodeV1(oldNode, newNode)
+							k.K8sEventProcessed(metricNode, metricUpdate, err == nil)
+						}
+					}
+				}
+				k.K8sEventReceived(metricNode, metricUpdate, valid, equal)
+			},
+		},
+		k8s.ConvertToNode,
```

Comment:
I think these are no longer needed given @aanm's latest changes but given it selects a single node, the resource waste is minimal.

Reply:
line 56 can be nil

Reply:
Done. I also had to change line 35 to use `slim_corev1`.